From 522d21154b8ee9a824d854c620e87fff1747b6dd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sat, 23 Mar 2013 01:29:23 +0100
Subject: [PATCH 3/3] loginDialog: Remove logo in upper left corner

With optional branding now being shown below the user list, we can
remove the unloved instance in the upper left corner ...

https://bugzilla.gnome.org/show_bug.cgi?id=694912
---
 js/gdm/loginDialog.js | 30 ------------------------------
 js/ui/panel.js        |  1 -
 js/ui/sessionMode.js  |  2 +-
 3 files changed, 1 insertion(+), 32 deletions(-)

diff --git a/js/gdm/loginDialog.js b/js/gdm/loginDialog.js
index bba17ae..b833f95 100644
--- a/js/gdm/loginDialog.js
+++ b/js/gdm/loginDialog.js
@@ -57,36 +57,6 @@ const WORK_SPINNER_ANIMATION_TIME = 0.3;
 
 let _loginDialog = null;
 
-const LogoMenuButton = new Lang.Class({
-    Name: 'LogoMenuButton',
-    Extends: PanelMenu.Button,
-
-    _init: function() {
-        this.parent(0.0, null, true);
-
-        this._settings = new Gio.Settings({ schema: GdmUtil.LOGIN_SCREEN_SCHEMA });
-        this._settings.connect('changed::' + GdmUtil.LOGO_KEY,
-                               Lang.bind(this, this._updateLogo));
-
-        this._iconBin = new St.Bin();
-        this.actor.add_actor(this._iconBin);
-
-        this._updateLogo();
-    },
-
-    _updateLogo: function() {
-        let path = this._settings.get_string(GdmUtil.LOGO_KEY);
-        let icon = null;
-
-        if (path) {
-            let file = Gio.file_new_for_path(path);
-            let cache = St.TextureCache.get_default();
-            icon = cache.load_uri_async(file.get_uri(), -1, _LOGO_ICON_HEIGHT);
-        }
-        this._iconBin.set_child(icon);
-    }
-});
-
 const UserListItem = new Lang.Class({
     Name: 'UserListItem',
 
diff --git a/js/ui/panel.js b/js/ui/panel.js
index 851b9af..0e17cdc 100644
--- a/js/ui/panel.js
+++ b/js/ui/panel.js
@@ -923,7 +923,6 @@ const PANEL_ITEM_IMPLEMENTATIONS = {
     'volume': imports.ui.status.volume.Indicator,
     'battery': imports.ui.status.power.Indicator,
     'lockScreen': imports.ui.status.lockScreenMenu.Indicator,
-    'logo': imports.gdm.loginDialog.LogoMenuButton,
     'keyboard': imports.ui.status.keyboard.InputSourceIndicator,
     'powerMenu': imports.gdm.powerMenu.PowerMenuButton,
     'userMenu': imports.ui.userMenu.UserMenuButton
diff --git a/js/ui/sessionMode.js b/js/ui/sessionMode.js
index b62e1ab..6ddf041 100644
--- a/js/ui/sessionMode.js
+++ b/js/ui/sessionMode.js
@@ -45,7 +45,7 @@ const _modes = {
         unlockDialog: imports.gdm.loginDialog.LoginDialog,
         components: ['polkitAgent'],
         panel: {
-            left: ['logo'],
+            left: [],
             center: ['dateMenu'],
             right: ['a11yGreeter', 'display', 'keyboard',
                     'volume', 'battery', 'powerMenu']
-- 
1.8.2.1

